Firmware/ConfigurationStore.cpp

```diff
@@ -269,12 +216,11 @@ bool Config_RetrieveSettings()
 {
   bool previous_settings_retrieved = true;
     char ver[4]=EEPROM_VERSION;
-    EEPROM_readData(reinterpret_cast<uint8_t*>(EEPROM_M500_base->version), reinterpret_cast<uint8_t*>(cs.version), sizeof(cs.version), "cs.version"); //read stored version
+    eeprom_read_block(reinterpret_cast<uint8_t*>(cs.version), reinterpret_cast<uint8_t*>(EEPROM_M500_base->version), sizeof(cs.version));
```

Comment:
we could read the version as a dword if that saves something ... it is just 4B long anyway.

Reply:
I don't think that's worth it. I prefer not messing too much with the dumpster fire that is the CS. In any case, a lot of stuff needs to be retested after these changes

Reply:
ok